packages/shared/lib/src/sdk/transaction.rs

```diff
@@ -127,23 +128,29 @@ impl TransactionKind {
             TransactionKind::RevealPk(public_key) => {
                 let reveal_pk = RevealPkMsg::new(public_key.to_string());
                 borsh::to_vec(&reveal_pk)?
-            },
-            TransactionKind::TransparentTransfer(transparent_transfer) => {
-                let TransparentTransfer {
-                    amount,
-                    source,
-                    target,
-                    token,
-                } = transparent_transfer;
-
-                let transparent_transfer = TransparentTransferMsg::new(
-                    source.to_string(),
-                    target.to_string(),
-                    token.to_string(),
-                    amount.to_string(),
-                );
+            }
+            TransactionKind::Transfer(transfer) => {
+                let Transfer {
+                    sources,
+                    targets,
+                    shielded_section_hash: _,
+                } = transfer;
+
+                let mut data: Vec<TransferDataMsg> = vec![];
+
+                for (i, (source, amount)) in sources.into_iter().enumerate() {
+                    let (target, token) = targets.iter().nth(i).unwrap();
```

Comment:
It looks like `token` here is the amount, which is putting the amount in the token field when viewing the data.

Reply:
I think you're right, I must have messed it up at some point :(

Reply:
Updated, thanks!

Reply:
I am still seeing this error with the latest changes. I think it's that `sources` is actually a map of account →
 amount, not account → token. There is a `token` field on account that seems to work though, although I'm not sure if it should be the source account or the target account.